packages/fast-components-react-base/src/tabs/tabs.tsx

```diff
@@ -0,0 +1,313 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import { IManagedClasses, ITabsClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import Foundation, { HandledProps } from "../foundation";
+import { ITabsHandledProps, ITabsManagedClasses, ITabsUnhandledProps, TabsProps } from "./tabs.props";
+import Tab from "./tab";
+import TabItem from "./tab-item";
+import TabPanel from "./tab-panel";
+
+export enum TabSlot {
+    tab = "tab",
+    tabItem = "tab-item",
+    tabPanel = "tab-panel"
+}
+
+export interface ITabsState {
+    activeId: string;
+}
+
+class Tabs extends Foundation<ITabsHandledProps & ITabsManagedClasses, ITabsUnhandledProps, ITabsState> {
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(nextProps: TabsProps, prevState: ITabsState): null | ITabsState {
+        if (nextProps.activeId && nextProps.activeId !== prevState.activeId) {
+            return {
+                activeId: nextProps.activeId
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<ITabsHandledProps & IManagedClasses<ITabsClassNameContract>> = {
+        activeId: void 0,
+        children: void 0,
+        label: void 0,
+        managedClasses: void 0,
+        orientation: void 0
+    };
+
+    /**
+     * React ref for the tab list
+     */
+    private tabListRef: React.RefObject<HTMLDivElement>;
+
+    constructor(props: TabsProps) {
+        super(props);
+
+        const tabItems: JSX.Element[] = this.getChildBySlot(this.props.children, TabSlot.tabItem);
+        this.tabListRef = React.createRef();
+
+        this.state = {
+            activeId: this.props.activeId
+                ? this.props.activeId
+                : tabItems.length > 0
+                ? tabItems[0].props.id
+                : void(0)
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        const tabElements: JSX.Element[] = this.renderTabElements();
+
+        if (tabElements.length > 0) {
```

Comment:
I think I'm partial to always rendering the HTML that we can render. Deciding to render or not render is really an implementation concern - not a component concern.

I think I'd be a little confused if I dropped the component on a page but it didn't do anything. Its similar to the silent error scenario - I now have to know about the internal code logic to accurately debug. I don't think this is a major use case so I'm not too concerned about it, but I think I prefer to keep the components as simple as possible and leave implementation concerns to the authors.